.github/workflows/build.yml

```diff
@@ -0,0 +1,43 @@
+name: build
+on:
+    workflow_dispatch:
+    push:
+jobs:
+    build:
+        name: Build and upload to riffraff
+        runs-on: ubuntu-latest
+        # Allow GitHub to request an OIDC JWT ID token, to use with aws-actions/configure-aws-credentials
+        permissions:
+            id-token: write
+            contents: read
+        steps:
+            - name: Checkout the repository
+              uses: actions/checkout@v4
+
+            - name: Install Node
+              uses: actions/setup-node@v3
+              with:
+                  node-version-file: '.nvmrc'
+
+            - name: Install dependencies, build & run tests
+              run: |
+                npm install -g yarn
+                yarn
+                yarn build
+                yarn test
+
+            - name: Configure AWS credentials
+              uses: aws-actions/configure-aws-credentials@v3
```

Comment:
Let's use the [latest version](https://github.com/aws-actions/configure-aws-credentials/blob/main/CHANGELOG.md) of this action, v4.